StringVariable.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * StringVariable.h
+ *
+ *  Created on: Jun 20, 2018
+ *      Author: baiyuhuang
+ */
+
+#ifndef STRINGVARIABLE_H_
+#define STRINGVARIABLE_H_
+#include "StringVariable.h"
+#include "MessageConstants.h"
+#include <cstdlib>
+#include "stdio.h"
+#include "string.h"
+#include <typeinfo>
+
+const int DATA_STRING_SIZE = 11;
+
+class StringVariable {
+public:
+	StringVariable();
+    template <typename T>
+    void Serialize(T &input, char* buffer);
+    
+    void IntToString(int input, char* buffer);
+    void BoolToString(bool input, char* buffer);
+    template<typename T>
+    void FloatAndDoubleToString(T& input, char* buffer);
+private:
+    
+};
+
+//serialize float or double to string
+template <typename T>
+void StringVariable::FloatAndDoubleToString(T& input, char* buffer)
+{
+    if(typeid(input) == typeid(double) || typeid(input) == typeid(float)){
+        snprintf(buffer, sizeof(buffer), "%f", input);
+    }
+}
+
+//input a variable and it will serialize it
+template <typename T>
+void StringVariable::Serialize(T &input, char* buffer)
+{
+    *buffer = '\0';
```

Comment:
You can do: 
`buffer[0] = sp::START_CHAR;`
`buffer[1] = '\0';`
It's easier to read this way.